src/detours.cpp

```diff
@@ -1337,9 +1343,55 @@ static LONG                 s_nPendingError         = NO_ERROR;
 static PVOID *              s_ppPendingError        = NULL;
 static DetourThread *       s_pPendingThreads       = NULL;
 static DetourOperation *    s_pPendingOperations    = NULL;
+static MiniheapHeader  *    s_pMiniheap             = NULL;
 
 //////////////////////////////////////////////////////////////////////////////
 //
+#define ALIGN_ADDRESS(PTR, ALIGN) \
+    (((ULONG_PTR)(PTR) + (ALIGN) - 1) & ~((ULONG_PTR)(ALIGN) - 1))
+
+#define MIN(A, B) ((A) < (B) ? (A) : (B))
+
+#define NEW(TYPE) ((TYPE*)detour_miniheap_alloc(sizeof(TYPE), alignof(TYPE)))
```

Comment:
A `static_cast` is sufficient for converting a `void*` to a specific pointer type.
```suggestion
#define NEW(TYPE) (static_cast<TYPE*>(detour_miniheap_alloc(sizeof(TYPE), alignof(TYPE))))
```

Reply:
Even though the file is `.cpp`, it is written completely in C-stle (only C-style cases, no c++ includes, not a single `::`, etc.)